File: Backend/socialmedia/DTOs/userProfileDtos/profilepostDto.cs
namespace socialmedia.DTOs.userProfileDtos
{
    public class profilepostDto
    {
        public int Id { get; set; }
        public string Content { get; set; }
        public DateTime CreatedAt { get; set; }
        public List<profileimpressionDto> Impressions { get; set; }
        public List<profilecommentDto> Comments { get; set; }
    }
}

File: Frontend/client/src/app/models/ProfilePostDto.ts
import { ProfileCommentDto } from "./ProfileCommentDto";
import { ProfileImpressionDto } from "./ProfileImpressionDto";

export interface ProfilePostDto {
  id: number;
  content: string;
  createdAt: string;
  impressions: ProfileImpressionDto[];
  comments: ProfileCommentDto[];
}
